src/main/java/roomescape/controller/ReservationApiController.java

```diff
@@ -0,0 +1,50 @@
+package roomescape.controller;
+
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import roomescape.dto.ReservationRequestDto;
+import roomescape.dto.ReservationResponseDto;
+import roomescape.entity.ReservationEntity;
+
+@Controller
+@RequestMapping("/reservations")
+public class ReservationApiController {
+
+    private final List<ReservationEntity> reservationEntities;
+
+    public ReservationApiController(List<ReservationEntity> reservationEntities) {
+        this.reservationEntities = reservationEntities;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ReservationResponseDto>> getReservations() {
+        List<ReservationResponseDto> reservationResponseDtos = reservationEntities.stream()
+                .map(ReservationResponseDto::new)
+                .toList();
+
+        return new ResponseEntity<>(reservationResponseDtos, HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<ReservationResponseDto> createReservation(
+            @RequestBody ReservationRequestDto reservationRequestDto) {
+
+        ReservationEntity reservation = reservationRequestDto.toEntity();
+        reservationEntities.add(reservation);
+        return new ResponseEntity<>(new ReservationResponseDto(reservation), HttpStatus.OK);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteReservation(@PathVariable Long id) {
```

Comment:
```suggestion
    public ResponseEntity<Void> deleteReservation(@PathVariable long id) {
```

Long 과 long은 무슨 차이가 있을까요?

Reply:
음.. id에 long vs Long 양쪽 케이스를 모두 고려해 본다면

primitive type long은 `null`을 허용안하고 기본 값이 0이라, id 값이 0이어서 0인건지 null이어서 0인건지 구분이 될지 의문이 들고
wrapper class Long은 `null`을 허용하니 null이면 id 값이 없다는 것을 구분할 수 있지 않나 생각이 드네요.

id 말고 일반적인 데이터로 접근해보면, 설계에 따라 달라지겠지만 테이블에 데이터가 없는 칼럼은 보통 null 일 것 같은데
Long과 long을 섞어 쓰지만 않으면 괜찮지 않나 생각하고 있습니다.

Reply:
음.. id에 long vs Long 양쪽 케이스를 모두 고려해 본다면

primitive type long은 null을 허용안하고 기본 값이 0이라, id 값이 0이어서 0인건지 null이어서 0인건지 구분이 될지 의문이 들고
wrapper class Long은 null을 허용하니 null이면 id 값이 없다는 것을 구분할 수 있지 않나 생각이 드네요.

id 말고 일반적인 데이터로 접근해봤을 때, 
설계에 따라 달라지겠지만 테이블에 데이터가 없는 칼럼은 보통 null이 자연스럽다고 인식할 것 같은데,
long 때문에 0이 들어가게 되면 많이 어색하지 않을까 생각이듭니다.

Reply:
http request에 pathVariable id에 null을 허용하고 싶은 의도가 있는건가요?

Reply:
"테이블"이란 용어는 무엇인가요? 혹시 디비 테이블인가요? 프로젝트 어디에도 디비 관련을 찾을수 없는것 같아요